contracts/axelar-interchain-token-service/src/contract.rs

```diff
@@ -42,59 +39,52 @@ impl InterchainTokenService {
         event::transfer_ownership(&env, owner, new_owner);
     }
 
-    pub fn is_trusted_address(env: &Env, chain: String, address: String) -> bool {
-        let key = DataKey::TrustedAddress(TrustedItsAddress { chain_name: chain, trusted_address: address });
-        env.storage().persistent().has(&key)
+    pub fn trusted_address(env: &Env, chain: String) -> Option<String> {
+        env.storage()
+            .persistent()
+            .get(&DataKey::TrustedAddress(chain))
     }
 
     pub fn set_trusted_address(
-        env: Env, 
-        chain: String, 
+        env: Env,
+        chain: String,
         address: String,
-    ) -> Result<(), InterchainTokenServiceError> {
+    ) -> Result<(), ContractError> {
         Self::owner(&env).require_auth();
 
-        let key = DataKey::TrustedAddress(
-            TrustedItsAddress { 
-                chain_name: chain.clone(), 
-                trusted_address: address.clone()
-            }
-        );
+        let key = DataKey::TrustedAddress(chain.clone());
 
         ensure!(
             !env.storage().persistent().has(&key),
-            InterchainTokenServiceError::TrustedAddressAlreadyAdded
+            ContractError::TrustedAddressAlreadySet
         );
 
-        env.storage().persistent().set(&key, &true);
+        env.storage().persistent().set(&key, &address);
 
         event::set_trusted_address(&env, chain, address);
+
         Ok(())
-    } 
+    }
 
     pub fn remove_trusted_address(
-        env: Env, 
-        chain: String, 
+        env: Env,
+        chain: String,
         address: String,
```

Comment:
When removing a trusted address, only the chain (i.e key) needs to be passed in. See the EVM ITS methods for reference.
https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/interfaces/IAddressTracker.sol#L12